perf-event/src/events/msr.rs

```diff
@@ -0,0 +1,68 @@
+use crate::events::Event;
+use c_enum::c_enum;
+use perf_event_open_sys::bindings;
+use std::io;
+use std::sync::atomic::{AtomicU32, Ordering};
+
+// 0 will never be the PMU value for msr.
+// We use it as a flag value to indicate that this has not been initialized.
+static MSR_TYPE: AtomicU32 = AtomicU32::new(0);
+
+
+c_enum! {
+    /// MSR events on all x86 CPUs
+    #[derive(Clone, Copy, Eq, PartialEq, Hash)]
+    pub enum MSRConfig: u64 {
```

Comment:
There seems to be a number of other MSR ids supported by the kernel that aren't listed here. Could you add the rest as well?

https://sourcegraph.com/github.com/torvalds/linux/-/blob/arch/x86/events/msr.c?L8:6